stackrox/stackrox
# chore(ui): Add links to configmanagement for keyboard accessibility

### Description

### Problem from axe DevTools

> Scrollable region must have keyboard access `<div class="rt-table" role="grid">`

### Analysis

https://dequeuniversity.com/rules/axe/4.9/scrollable-region-focusable?application=AxeChrome

Classic routes that render tables have `onRowClick` event handler:
* which does not have associated keyboard action
* which is probably invisible to screen readers

### Solution

1. Add in name column a link to open the side panel.
2. Rewrite some files in TypeScript especially to declare entity types.

### Residue

1. After we finish this batch of accessiblity issues, adjust style rules to render links with PatternFly style in classic tables.
2. Investigate whether integration tests can replace click on row with click on link.

### User-facing documentation


### Testing


#### Automated testing


#### How I validated my change

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js 0 = 4637305 - 4637305
        total 925 = 11778625 - 11777700
    * `ls -al build/static/js/*.js | wc`
        files 0 = 174 - 174
3. `yarn start` in ui

### Manual testing

1. Visit /main/configmanagement, click **Scan** in **CIS Kubernetes 1.5** widget, and then click controls count link.

    * Before changes, see presence of accessibility issue.
        See tab key navigation skips the table.
        ![configmanagement_controls_with_issue](https://github.com/user-attachments/assets/0cd485b4-828f-46c4-82da-92e983f70892)

    * After changes, see absence of accessibility issue.
        See tab key navigation visits name link in each row of the table.
        Press enter key to open the side panel.
        ![configmanagement_controls_with_link](https://github.com/user-attachments/assets/cead69cf-0d88-4b6c-929f-c86be32c69a2)

2. Visit /main/configmanagement/clusters and after changes, see tab key navigation.
    ![configmanagement_clusters_with_link](https://github.com/user-attachments/assets/39baffb8-19c0-4d38-9844-6e5f8b2149ea)

3. Visit /main/configmanagement/deployments and after changes, see tab key navigation.
    ![configmanagement_deployments_with_link](https://github.com/user-attachments/assets/5056ffd9-0cfa-4f10-a7a4-6c06b5a70cad)

4. Visit /main/configmanagement/images and after changes, see tab key navigation.
    ![configmanagement_images_with_link](https://github.com/user-attachments/assets/df5b9fca-e037-492e-9cd1-b5040ca8735c)

5. Visit /main/configmanagement/namespaces and after changes, see tab key navigation.
    ![configmanagement_namespaces_with_link](https://github.com/user-attachments/assets/fe844836-9d88-4886-883a-3a82b787e5c4)

6. Visit /main/configmanagement/nodes and after changes, see tab key navigation.
    ![configmanagement_nodes_with_link](https://github.com/user-attachments/assets/db147ca8-f07c-4cf5-b682-039685959d6e)

7. Visit /main/configmanagement/policies.

    * Before changes, see presence of accessibility issue.
        See tab key navigation skips the table.
        ![configmanagement_policies_with_issue](https://github.com/user-attachments/assets/7fd130d2-1c86-450f-a6e1-af55e09ac7be)

    * After changes, see absence of accessibility issue.
        See tab key navigation visits name link in each row of the table.
        Press enter key to open the side panel.
        ![configmanagement_policies_with_link](https://github.com/user-attachments/assets/2b68cf2e-bd5b-4569-a546-0d3716e3137f)

8. Visit /main/configmanagement/roles and after changes, see tab key navigation.
    ![configmanagement_roles_with_link](https://github.com/user-attachments/assets/a5195252-12ba-459a-ba41-9cb6f38aa173)

9. Visit /main/configmanagement/secrets and after changes, see tab key navigation.
    ![configmanagement_secrets_with_link](https://github.com/user-attachments/assets/87e4f069-7711-4a6a-ac66-4b0b45fca232)

10. Visit /main/configmanagement/serviceaccounts and after changes, see tab key navigation.
    ![configmanagement_serviceaccounts_with_link](https://github.com/user-attachments/assets/c0bc0803-d918-4b1c-bd8e-1f77e9bca377)

11. Visit /main/configmanagement/subjects and after changes, see tab key navigation.
    ![configmanagement_subjects_with_link](https://github.com/user-attachments/assets/3bb32b98-3554-479a-aa85-dce0c965d31e)
